src/mito2/src/sst/parquet/row_group.rs

```diff
@@ -59,6 +63,8 @@ impl<'a> InMemoryRowGroup<'a> {
         parquet_meta: &'a ParquetMetaData,
         row_group_idx: usize,
         cache_manager: Option<CacheManagerRef>,
+        file_path: &'a str,
+        object_store: ObjectStore,
```

Comment:
Can we keep `InMemoryRowGroup` in memory?

We can add a wrapper that implements `AsyncFileReader` to do concurrent fetch.

Reply:
> I suggest adding a wrapper that implements `AsyncFileReader` to do concurrent fetch.

`AsyncFileReader` also requires to implement the [get_metadata()](https://docs.rs/parquet/latest/parquet/arrow/async_reader/trait.AsyncFileReader.html#tymethod.get_metadata) method, which the `InMemoryRowGroup` doesn't use.

Reply:
For the wrapper, to implement `get_metadata` can just redirect to the inner reader. Not a big obstacle.

Reply:
We want to get rid of the creation of the underlying reader and also remove the dependency of the `AsyncFileReader` trait.

> to implement get_metadata can just redirect to the inner reader.

Creating such a reader for the full range will fetch the whole object from the object store and that's what I want to get rid of.

Reply:
> We want to get rid of the creation of the underlying reader
Got it. I didn't think about that behavior before.